Tests/Common/TestOutputLogger.cs

```diff
@@ -35,7 +35,16 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             if (!IsEnabled(logLevel)) return;
 
             var message = formatter(state, exception);
-            this.testOutputHelper.WriteLine(message);
+
+            try
+            {
+                this.testOutputHelper.WriteLine(message);
+            }
+            catch (Exception)
```

Comment:
What about making this specific to `InvalidOperationException`, which is the [quoted case](https://github.com/Azure/iotedge-lorawan-starterkit/issues/1554) (for now)?

```suggestion
            catch (InvalidOperationException)
```